numpy/core/src/multiarray/compiled_base.c

```diff
@@ -795,7 +795,52 @@ static int sequence_to_arrays(PyObject *seq,
     return 0;
 }
 
-/* Inner loop for unravel_index */
+/*
+ * Force-cast an array into intp if it is empty. Fail otherwise.
+ *
+ * op        - Pointer to the array pointer. Will be modified if
+ *             recasting is needed.
+ * paramname - Name to be used for error string.
+ */
+static int _cast_to_intp_or_fail(PyArrayObject **op,
+                                 char *paramname)
+{
+    PyArray_Descr *dtype = PyArray_DescrFromType(NPY_INTP);
+    if (dtype == NULL) {
+        return -1;
+    }
+
+    if (!PyArray_CanCastArrayTo(*op, dtype, NPY_SAME_KIND_CASTING)) {
```

Comment:
ping?

On Thu, Feb 4, 2016 at 5:16 PM, Jaime <notifications@github.com> wrote:

> In numpy/core/src/multiarray/compiled_base.c
> <https://github.com/numpy/numpy/pull/7170#discussion_r51946835>:
>
> > +/*
> > + * Force-cast an array into intp if it is empty. Fail otherwise.
> > + *
> > + * op        - Pointer to the array pointer. Will be modified if
> > + *             recasting is needed.
> > + * paramname - Name to be used for error string.
> > + */
> > +static int _cast_to_intp_or_fail(PyArrayObject **op,
> > +                                 char *paramname)
> > +{
> > +    PyArray_Descr *dtype = PyArray_DescrFromType(NPY_INTP);
> > +    if (dtype == NULL) {
> > +        return -1;
> > +    }
> > +
> > +    if (!PyArray_CanCastArrayTo(*op, dtype, NPY_SAME_KIND_CASTING)) {
>
> You are absolutely right, and I should not look at any more code today.
> Sorry for the noise...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/numpy/numpy/pull/7170/files#r51946835>.
>